### DISCUSSION TITLE
Best way to handle several results files

### DISCUSSION THREAD

**Discussion by crile** [allure-framework/allure2#1433] [Questions & Support]
Hi,

I work with several projects, so I put every xml reports in the directory then I generate allure report.

Everything is great until all xml reports are not present at the same time.

Example:

1. I start with two reports

    * `junit1.xml`, testsuite 1, testcase 1
    * `junit2.xml`, testsuite 2, testcase 1

2. I run `allure generate` command
3. I copy a new `junit1.xml` linked to the last execution of my tests
4. I run `allure generate` command
  
    * I open the report, and I "_loss_" the results from the `junit2.xml`

5. I copy a new `junit2.xml` linked to the last execution of my tests
6. I run `allure generate` command

    * I open the report, and I "_loss_" the results from the `junit1.xml`

Can I have some sort of history with `not run` status for all testcases that missing since the report is not present ?
Any idea to handle my use case ? As many instances of allure report than xml results ?

I have five xml reports with their own execution frequency.

I wish I will be able to keep the global view.

Thanks

**Comment by Avaddon-tfa** [comment]
Well, such behavior might be a bug. 

Would you mind creating a repo/file example with the example so we would be able to reproduce it on our side?

**Comment by crile** [comment]
To reproduce the problem run those commands:

```
make case1
make case2
make case3
```


makefile
``` 
RESULTS        := results
ALLURE         := $(shell which allure)

case1:
	rm -fr $(RESULTS)
	mkdir $(RESULTS)
	cp template/* $(RESULTS)
	$(ALLURE) generate --clean $(RESULTS)/
	$(ALLURE) open

case2:
	rm -fr $(RESULTS)
	mkdir $(RESULTS)
	cp -R allure-report/history $(RESULTS)
	cp template/junit1.xml $(RESULTS)
	$(ALLURE) generate --clean $(RESULTS)/
	$(ALLURE) open

case3:
	rm -fr $(RESULTS)
	mkdir $(RESULTS)
	cp -R allure-report/history $(RESULTS)
	cp template/junit2.xml $(RESULTS)
	$(ALLURE) generate --clean $(RESULTS)/
	$(ALLURE) open


.PHONY: case1 case2 case3
```

junit1.xml
```
<?xml version="1.0" ?>
<testsuites disabled="0" errors="0" failures="0" tests="2" time="0.000301361083984375">
	<testsuite disabled="0" errors="0" failures="0" name="junit1" skipped="0" tests="2" time="0.000301361083984375">
		<testcase name="my_fixture" time="0.000163" classname="TestSomeThing" class="fixture"/>
		<testcase name="nested_case" time="0.000034" classname="module_name.test" class="function"/>
	</testsuite>
</testsuites>
```

junit2.xml
```
<?xml version="1.0" ?>
<testsuites disabled="0" errors="0" failures="0" tests="1" time="0.000301361083984375">
	<testsuite disabled="0" errors="0" failures="1" name="junit2" skipped="0" tests="1" time="0.000301361083984375">
		<testcase name="testcase1" time="0.000163" classname="testcase1" class="fixture"/>
	</testsuite>
</testsuites>
```

**Reply by Avaddon-tfa** [reply]
Thank you for the details and sorry for the delay. 
We are working on it :)
